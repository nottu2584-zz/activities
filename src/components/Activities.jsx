import { default as React } from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  activities: {
    background: "#F3F4F6",
    color: "#282c34",
    borderRadius: 5,
    fontSize: "0.9em",
    overflow: "hidden",
    padding: 0,
  },
  row: {
    display: "flex",
    flexWrap: "wrap",
    width: "100%",
    "&.head": {
      borderBottom: "1px solid #DCDFE5",
      fontFamily: "'Rubik', sans-serif",
      fontSize: "1.2em",
      fontWeight: "bold",
      textTransform: "uppercase",
    },
    "&:nth-child(even)": {
      backgroundColor: "#DCDFE5",
    },
  },
  cell: {
    boxSizing: "border-box",
    flexGrow: "1",
    padding: "0.8em 1.2em",
    overflow: "hidden",
    listStyle: "none",
    "&.col-1": {
      flex: "0 0 50px",
    },
    "&.col-2": {
      flex: "0 0 calc(50% - 50px)",
      textAlign: "left",
    },
    "&.col-3": {
      flex: "0 0 25%",
    },
    "&.col-4": {
      flex: "0 0 25%",
    },
  },
});

const Activities = (props) => {
  const classes = useStyles();
  const { activities, checked, handleOnChange } = props;

  return (
    <div className={`activities ${classes.activities}`}>
      <div className={`${classes.row} row head`}>
        <div className={`${classes.cell} col col-1`}></div>
        <div className={`${classes.cell} col col-2`}>Activity Name</div>
        <div className={`${classes.cell} col col-3`}>COVID Friendly</div>
        <div className={`${classes.cell} col col-4`}>Times Performed</div>
      </div>
      {activities
        ? activities.map((activity, index) => {
            return (
              <div key={index} className={`${classes.row} row row-${index}`}>
                <div className={`${classes.cell} col col-1`}>
                  <input
                    name={`activity-${activity.id}`}
                    id={`activity-${activity.id}`}
                    type="checkbox"
                    checked={checked.some(
                      (item) => item.id === activity.id
                    )}
                    onChange={(e) => handleOnChange(e, activity)}
                  />
                </div>
                <div className={`${classes.cell} col col-2`}>
                  {activity.name}
                </div>
                <div className={`${classes.cell} col col-3`}>
                  {activity.friendly ? "Yes" : "No"}
                </div>
                <div className={`${classes.cell} col col-4`}>
                  {activity.performed}
                </div>
              </div>
            );
          })
        : null}
    </div>
  );
};

export default Activities;
