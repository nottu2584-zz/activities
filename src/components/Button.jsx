
import { default as React } from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  button: {
    background: "#009b72",
    boxShadow: "0 1px 2px #333",
    border: "none",
    borderRadius: "5px",
    color: "#ffffff",
    cursor: "pointer",
    fontFamily: "'Rubik', sans-serif",
    fontSize: "1.1em",
    lineHeight: "1.1em",
    outline: "none",
    padding: "1em 1.25em",
    MozTransition: "box-shadow 0.25s",
    OTransition: "box-shadow 0.25s",
    WebkitTransition: "box-shadow 0.25s",
    transition: "box-shadow 0.25s",
    WebkitFontSmoothing: "antialiased",
    MozOsxFontSmoothing: "grayscale",
  },
  "&:hover": { boxShadow: "0 1px 20px #333" },
});

const Button = (props) => {
  const classes = useStyles();
  
  const {handleOnClick} = props;

  return (
    <button className={classes.button} onClick={(e) => handleOnClick(e)}>
      {props.children}
    </button>
  );
};

export default Button;