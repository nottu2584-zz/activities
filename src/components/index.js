export { default as Activities } from "./Activities";
export { default as Button } from "./Button";
export { default as Modal } from "./Modal";

