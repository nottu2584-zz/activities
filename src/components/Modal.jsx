import { createUseStyles } from "react-jss";
import Button from "./Button";

const useStyles = createUseStyles({
  modal: {
    backgroundColor: "rgba(0, 0, 0, 0.85)",
    position: "fixed",
    height: "100%",
    width: "100%",
    bottom: 0,
    left: 0,
    right: 0,
    top: 0,
    zIndex: 1,
  },
  wrapper: {
    display: "flex",
    flexDirection: "column",
    flexWrap: "wrap",
    alignItems: "center",
    height: "100%",
    justifyContent: "center",
    zIndex: 2,
  },
  content: {
    display: "inline-flex",
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center",
    maxWidth: "30%",
    textAlign: "center",
    "& p": {
      margin: "1em 0 0",
    },
    "& ul": {
      textAlign: "left",
      margin: "1em 0",
    },
  },
  fullWidth: {
    flex: "1 1 100%",
  },
  button: {
    background: "#009B72",
    boxShadow: "0 1px 2px #333",
    border: "none",
    borderRadius: 5,
    color: "#FFFFFF",
    cursor: "pointer",
    fontSize: "1.25em",
    lineHeight: "1.1em",
    outline: "none",
    padding: "1em 1.25em",
    "-moz-transition": "box-shadow 0.25s",
    "-o-transition": "box-shadow 0.25s",
    "-webkit-transition": "box-shadow 0.25s",
    transition: "box-shadow 0.25s",
    "&:hover": {
      boxShadow: "0 1px 20px #333",
    },
  },
});

const Modal = (props) => {
  const classes = useStyles();

  const { activities, handleModal, message, title } = props;

  return (
    <div className={`modal ${classes.modal}`}>
      <div className={`wrapper ${classes.wrapper}`}>
        <div className={`content ${classes.content}`}>
          <h2 className={classes.fullWidth}>{title}</h2>
          <p className={classes.fullWidth}>{message}</p>
          <ul className={classes.fullWidth}>
            {activities
              ? activities.map((activity, index) => (
                  <li key={index}>{activity.name}</li>
                ))
              : null}
          </ul>
          <Button handleOnClick={handleModal}>Oh, I see!</Button>
        </div>
      </div>
    </div>
  );
};

export default Modal;
