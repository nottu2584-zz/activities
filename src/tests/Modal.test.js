import { shallow } from "enzyme";
import { Button, Modal } from "../components";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useLayoutEffect: jest.requireActual("react").useEffect,
}));

describe("<Modal /> without props", () => {
  const container = shallow(<Modal />);
  it("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  it("should have an some wrappers", () => {
    expect(container.find(".modal").length).toEqual(1);
    expect(container.find(".wrapper").length).toEqual(1);
    expect(container.find(".content").length).toEqual(1);
  });

  it("should have an some content", () => {
    expect(container.find("h2").length).toEqual(1);
    expect(container.find("p").length).toEqual(1);
    expect(container.find("ul").length).toEqual(1);
  });

  it("should have a button", () => {
    expect(container.find(Button).length).toEqual(1);
  });
});

describe("<Modal /> with props", () => {
  const initialProps = {
    activities: [
      {
        id: 3,
        name: "Vacations in Hawaii",
        friendly: false,
        performed: 1,
      },
    ],
    handleModal: jest.fn(),
    title: 'some_title',
    message: 'some_message'
  };
  const container = shallow(<Modal {...initialProps} />);

  it("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  it("should have an some wrappers", () => {
    expect(container.find(".modal").length).toEqual(1);
    expect(container.find(".wrapper").length).toEqual(1);
    expect(container.find(".content").length).toEqual(1);
  });

  it("should have an some content", () => {
    expect(container.find("h2").length).toEqual(1);
    expect(container.find("p").length).toEqual(1);
    expect(container.find("ul").length).toEqual(1);
  });

  it("should have a button", () => {
    expect(container.find(Button).length).toEqual(1);
  });

  it("should have a title", () => {
    expect(container.find("h2").text()).toEqual(initialProps.title);
  });

  it("should have a message", () => {
    expect(container.find("p").text()).toEqual(initialProps.message);
  });

  it("should have an activities list", () => {
    container.find('li').forEach((node, index) => {
      expect(node.text()).toEqual(initialProps.activities[index].name);
    });
  });

  it("should have a <Button /> with a callback", () => {
      expect(container.find(Button).props().handleOnClick).toEqual(initialProps.handleModal);
  });
});
