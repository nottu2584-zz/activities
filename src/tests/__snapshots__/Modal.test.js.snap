// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> with props should match the snapshot 1`] = `"<div class=\\"modal modal-0-2-1\\"><div class=\\"wrapper wrapper-0-2-2\\"><div class=\\"content content-0-2-3\\"><h2 class=\\"fullWidth-0-2-4\\">some_title</h2><p class=\\"fullWidth-0-2-4\\">some_message</p><ul class=\\"fullWidth-0-2-4\\"><li>Vacations in Hawaii</li></ul><button class=\\"button-0-2-6\\">Oh, I see!</button></div></div></div>"`;

exports[`<Modal /> without props should match the snapshot 1`] = `"<div class=\\"modal modal-0-2-1\\"><div class=\\"wrapper wrapper-0-2-2\\"><div class=\\"content content-0-2-3\\"><h2 class=\\"fullWidth-0-2-4\\"></h2><p class=\\"fullWidth-0-2-4\\"></p><ul class=\\"fullWidth-0-2-4\\"></ul><button class=\\"button-0-2-6\\">Oh, I see!</button></div></div></div>"`;
